src/hotspot/share/opto/vtransform.cpp

```diff
@@ -866,28 +857,27 @@ VTransformApplyResult VTransformBoolVectorNode::apply(VTransformApplyState& appl
 }
 
 VTransformApplyResult VTransformReductionVectorNode::apply(VTransformApplyState& apply_state) const {
-  Node* first = nodes().at(0);
-  uint  vlen = nodes().length();
-  int   opc  = first->Opcode();
-  BasicType bt = first->bottom_type()->basic_type();
+  int     sopc = scalar_opcode();
+  uint    vlen = vector_length();
+  BasicType bt = element_basic_type();
 
   Node* init = apply_state.transformed_node(in_req(1));
   Node* vec  = apply_state.transformed_node(in_req(2));
 
-  ReductionNode* vn = ReductionNode::make(opc, nullptr, init, vec, bt);
+  ReductionNode* vn = ReductionNode::make(sopc, nullptr, init, vec, bt);
   register_new_node_from_vectorization_and_replace_scalar_nodes(apply_state, vn);
   return VTransformApplyResult::make_vector(vn, vlen, vn->vect_type()->length_in_bytes());
 }
 
 VTransformApplyResult VTransformLoadVectorNode::apply(VTransformApplyState& apply_state) const {
+  int     sopc = scalar_opcode();
+  uint    vlen = vector_length();
+  BasicType bt = element_basic_type();
   LoadNode* first = nodes().at(0)->as_Load();
-  uint  vlen = nodes().length();
-  Node* ctrl = first->in(MemNode::Control);
+  Node* ctrl = apply_state.transformed_node(in_req(MemNode::Control));
+  // first has the correct memory state, determined by VTransformGraph::apply_memops_reordering_with_schedule
   Node* mem  = first->in(MemNode::Memory);
-  Node* adr  = first->in(MemNode::Address);
-  int   opc  = first->Opcode();
-  const TypePtr* adr_type = first->adr_type();
-  BasicType bt = apply_state.vloop_analyzer().types().velt_basic_type(first);
+  Node* adr  = apply_state.transformed_node(in_req(MemNode::Address));
```

Comment:
There is still minimal reliance on `nodes` / `first`: but only for `mem` state. And currently, we cannot remove that yet, because we rely on the memory graph being reordered before vectorization, see `VTransformGraph::apply_memops_reordering_with_schedule`.

In a future RFE, I will refactor scheduling, so that we build the memory graph during apply.
See step 3 in [plan overfiew.](https://bugs.openjdk.org/browse/JDK-8340093)